### PR TITLE
Database from diagram

### PR DESCRIPTION
**Pull request by sameedmohsin28**
In this pull request, we have

Implemented the database from the diagram given in the project requirements.
Joined tables with relationships.
Added the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,61 @@
+-- create patients table
+CREATE TABLE patients (
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL,
```
Comment (Ghiftee) [inline schema_based_on_diagram.sql]: - [ ] Kindly ensure you use the same data type shown in the ERD. `Name` should be `varchar` and not `text`. Deviating from the ERD can introduce inconsistencies and make it harder to understand and maintain the database structure.

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  date_of_birth DATE NOT NULL
+);
+
+-- create medical_histories table
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at TIMESTAMP NOT NULL,
+  status TEXT NOT NULL,
```
Comment (Ghiftee) [inline schema_based_on_diagram.sql]: - [ ] Kindly ensure you use the same data type shown in the ERD. `Status` should be `varchar` and not `text`. Deviating from the ERD can introduce inconsistencies and make it harder to understand and maintain the database structure.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  patient_id INTEGER NOT NULL,
+  FOREIGN KEY (patient_id) REFERENCES patients (id)
+);
+
+-- create treatments table
+CREATE TABLE treatments (
+  id SERIAL PRIMARY KEY,
+  type TEXT NOT NULL,
+  name TEXT NOT NULL
```
Comment (Ghiftee) [inline schema_based_on_diagram.sql]: - [ ] Kindly ensure you use the same data type shown in the ERD. `Name` and `Type` should be `varchar` and not `text`. Deviating from the ERD can introduce inconsistencies and make it harder to understand and maintain the database structure.

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  treatment_id INTEGER NOT NULL,
+  medical_history_id INTEGER NOT NULL,
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id),
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+-- create invoices table
+CREATE TABLE invoices (
+  id SERIAL PRIMARY KEY,
+  total_amount NUMERIC(10, 2) NOT NULL,
```
Comment (Ghiftee) [inline schema_based_on_diagram.sql]: - [ ] Kindly ensure you use the same data type shown in the ERD. `total_amount` should be `decimal` and not `numeric`. Deviating from the ERD can introduce inconsistencies and make it harder to understand and maintain the database structure.

File: schema_based_on_diagram.sql [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  medical_history_id INTEGER NOT NULL,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+-- create invoice_items table for many-to-many relationship
+CREATE TABLE invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price NUMERIC(10,2) NOT NULL,
+  quantity INTEGER NOT NULL,
+  total_price NUMERIC(10,2) NOT NULL,
```
Comment (Ghiftee) [inline schema_based_on_diagram.sql]: - [ ] Kindly ensure you use the same data type shown in the ERD. `unit_price` and `total_price` should be `decimal` and not `numeric`. Deviating from the ERD can introduce inconsistencies and make it harder to understand and maintain the database structure.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @sameedmohsin28, @yosaddis  and @Cosmosha 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Tables are created using `Create table`
✔️ Foreign keys are set
✔️ Join tables are created

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._
- [ ]  When I try to view your license, I get this error, please fix it as public repositories on GitHub are often used to share open-source software. For your repository to truly be open source, you'll need to license it so that others are free to use, change, and distribute the software.

Screenshot 2023-06-02 at 21 01 37 (2)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewer

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**